src/Components/Server/src/Builder/RazorComponentsEndpointBuilderExtensions.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Components.Server;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extensions for <see cref="RazorComponentsEndpointBuilder"/>.
+    /// </summary>
+    public static class ComponentsEndpointbuilderExtensions
+    {
+        /// <summary>
+        /// Adds <typeparamref name="TComponent"/> to the list of components registered with this <see cref="ComponentsHub"/> instance.
+        /// The DOM selector associated with the <typeparamref name="TComponent"/> will default to the component type name in lowercase.
+        /// </summary>
+        /// <typeparam name="TComponent">The component type.</typeparam>
+        /// <param name="builder">The <see cref="RazorComponentsEndpointBuilder"/>.</param>
+        /// <returns>The <paramref name="builder"/>.</returns>
+        public static RazorComponentsEndpointBuilder AddComponent<TComponent>(this RazorComponentsEndpointBuilder builder)
```

Comment:
do we think that lowercasing the component name is a good convention that's obvious?

I feel like using `app` is already a convention that we've built but I'm unsure about the other convention. I'm generally in favor of being explicit in all but the default scenario.

Reply:
I guess I'm questioning if we should have this overload at all.

Reply:
I believe that an all lowercase "tag like" convention will be fine and understood. In the end, the casing doesn't matter for the selector, but only so that when you put it in code it doesn't get recognized as a Component by the parser.

Reply:
Based on the conversations we had I'll kill the selectorless overloads

Reply:
Done